design/cdk-bootstrap.md

```diff
@@ -0,0 +1,444 @@
+                      "Fn::Sub": "${StagingBucket.Arn}"
+                    },
+                    {
+                      "Fn::Sub": "${StagingBucket.Arn}/*"
+                    }
+                  ]
+                },
+                {
+                  "Action": [
+                    "ecr:PutImage",        "ecr:InitiateLayerUpload",
+                    "ecr:UploadLayerPart", "ecr:CompleteLayerUpload"
+                  ],
+                  "Resource": {
+                    "Fn::Sub": "${DockerAssetsStagingRepository.Arn}"
+                  }
+                }
+              ]
+            },
+            "PolicyName": {
+              "Fn::Sub": "DefaultBootstrapPublishRolePolicy-${AWS::Region}"
+            }
+          }
+        ],
+        "RoleName": "some-name"
+      }
+    },
+    "AssumedActionRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Ref": "TrustedAccounts"
+                }
+              }
+            }
+          ]
+        },
+        "Policies": [
+          {
+            "PolicyDocument": {
+              "Statement": [
+                {
+                  "Action": [
+                    "cloudformation:CreateChangeSet",   "cloudformation:DeleteChangeSet",
+                    "cloudformation:DescribeChangeSet", "cloudformation:DescribeStacks",
+                    "cloudformation:ExecuteChangeSet",
+                    "s3:GetObject*",                    "s3:GetBucket*",
+                    "s3:List*",                         "s3:Abort*",
+                    "s3:DeleteObject*",                 "s3:PutObject*",
+                    "kms:Decrypt",                      "kms:DescribeKey"
+                  ],
+                  "Resource": "*"
+                },
+                {
+                  "Action": "iam:PassRole",
+                  "Resource": {
+                    "Fn::Sub": "${CfnDeploymentRole.Arn}"
+                  }
+                }
+              ]
+            },
+            "PolicyName": {
+              "Fn::Sub": "DefaultBootstrapAssumedActionRolePolicy-${AWS::Region}"
+            }
+          }
+        ],
+        "Condition": "HasTrustedAccounts"
+      }
+    },
+    "CfnDeploymentRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Effect": "Allow",
+              "Principal": {
+                "Service": "cloudformation.amazonaws.com"
+              }
+            }
+          ]
+        },
+        "ManagedPolicyArns": {
+          "Ref": "CfnDeploymentRoleManagedPolicies"
+        },
+        "Condition": "HasTrustedAccounts"
+      }
+    }
+  },
+  "Outputs": {
```

Comment:
Is this needed for backwards compatibility with old CLI?

Let’s discuss our backwards compatibility design into a separate section. I am not sure this will be sufficient.

Reply:
Yes, it's for backwards compatibility. Do you see any problem with leaving them?

Reply:
If there's a way to actually utilize them for backwards compatibility, then I think it makes sense to leave them. I wonder if they fall under [this case](https://github.com/aws/aws-cdk/pull/3437/files#diff-de4732ee7cda1362f8a56569577cb7a0R616). What do you think?

Reply:
I believe these outputs are actually needed for multiple cases when the bootstrap stack is "new":

* old CLI, old framework, old init template
* old CLI, old framework, new init template
* old CLI, new framework, old init template
* new CLI, old framework, old init template
* new CLI, old framework, new init template

See the table I have in the newly added "Backwards compatibility" section in the newest revision.

Reply:
So basically we are saying that the new bootstrap stack will be compatible with old CLI/framework, correct? This needs to be explicitly stated in the doc.

Reply:
I've expanded on this section, as a table wasn't clear enough.